actions/actions.py

```diff
@@ -422,24 +462,77 @@ def name(self):
         return "action_account_balance"
 
     def run(self, dispatcher, tracker, domain):
-        init_account_balance = float(tracker.get_slot("account_balance"))
+        account_balance = float(tracker.get_slot("account_balance"))
         amount = tracker.get_slot("amount_transferred")
         if amount:
             amount = float(tracker.get_slot("amount_transferred"))
-            account_balance = init_account_balance - amount
+            init_account_balance = account_balance + amount
             dispatcher.utter_message(
                 template="utter_changed_account_balance",
                 init_account_balance=f"{init_account_balance:.2f}",
                 account_balance=f"{account_balance:.2f}",
             )
-            return [
-                SlotSet("payment_amount", None),
-                SlotSet("account_balance", account_balance),
-                SlotSet("amount_transferred", None),
-            ]
+            return [SlotSet("payment_amount", None)]
         else:
             dispatcher.utter_message(
                 template="utter_account_balance",
-                init_account_balance=f"{init_account_balance:.2f}",
+                init_account_balance=f"{account_balance:.2f}",
             )
             return [SlotSet("payment_amount", None)]
+
+
+class ActionCreditCardBalance(Action):
+    def name(self):
+        return "action_credit_card_balance"
+
+    def run(self, dispatcher, tracker, domain):
+        cc_balance = tracker.get_slot("credit_card_balance")
+        for cc in cc_balance.keys():
+            current_balance = cc_balance[cc]["current balance"]
+            dispatcher.utter_message(
+                template="utter_credit_card_balance",
+                credit_card=cc,
+                amount_of_money=f"{current_balance:.2f}",
+            )
+
+        return []
+
+
+class ActionSessionStart(Action):
+    def name(self) -> Text:
+        return "action_session_start"
```

Comment:
Actually, it may be that i'm supposed to send `/session_start` now instead. however the bot doesn't seem to like that:
```
Exception occurred while handling uri: 'http://localhost:5055/webhook'
Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/sanic/app.py", line 976, in handle_request
    response = await response
  File "/Users/ella/RasaHQ/rasa-sdk/rasa_sdk/endpoint.py", line 102, in webhook
    result = await executor.run(action_call)
  File "/Users/ella/RasaHQ/rasa-sdk/rasa_sdk/executor.py", line 385, in run
    events = await action(dispatcher, tracker, domain)
  File "/Users/ella/RasaHQ/financial-demo/actions/actions.py", line 525, in run
    events.extend(self._slot_set_events_from_tracker(tracker))
  File "/Users/ella/RasaHQ/financial-demo/actions/actions.py", line 511, in _slot_set_events_from_tracker
    for event in tracker.applied_events()
  File "/Users/ella/RasaHQ/financial-demo/actions/actions.py", line 512, in <listcomp>
    if isinstance(event, SlotSet)
TypeError: isinstance() arg 2 must be a type or tuple of types
```

can you look into that?

Reply:
Looks like it was a bug in my actions_session_start code, should be resolved now